qiskit_nature/transformers/second_quantization/electronic/seniority_zero_transformer.py

```diff
@@ -0,0 +1,242 @@
+
+            transformed_property.second_q_ops = partial(  # type: ignore[assignment]
+                second_q_ops_electronic_energy, transformed_property
+            )
+
+        elif isinstance(prop, ParticleNumber):
+            transformed_property = prop
+
+            transformed_property.second_q_ops = partial(  # type: ignore[assignment]
+                second_q_ops_particle_number, transformed_property
+            )
+        elif isinstance(prop, DriverMetadata):
+            transformed_property = prop
+        elif isinstance(prop, SecondQuantizedProperty):
+            transformed_property = None
+        elif isinstance(prop, ElectronicBasisTransform):
+            transformed_property = None
+
+        else:
+            raise TypeError(f"{type(prop)} is an unsupported Property-type for this Transformer!")
+
+        return transformed_property
+
+    def _restricted_electronic_integrals(
+        self, h_1: np.ndarray, h_2: np.ndarray
+    ) -> Tuple[np.ndarray, np.ndarray]:
+        """
+        Transform PyScf integrals to the restricted formalism
+        arxiv: 2002.00035, Appendix A, Equations (A10),(A11),(A12)
+        https://arxiv.org/pdf/2002.00035.pdf
+
+        Args:
+            h_1 (np.ndarray): one-body integrals, (N x N)
+            h_2 (np.ndarray): two-body integrals, (N x N x N x N)
+
+        Returns:
+            hr_1 (np.ndarray): restricted one-body integrals, (N x N)
+            hr_2 (np.ndarray): restricted two-body integrals, (N x N)
+        """
+        # change of notation (may only be valid for PySCF -> restricted formalism)
+        h_2 = np.einsum("ijkl->ljik", h_2)
+
+        hr_1 = np.zeros((self._num_modes, self._num_modes))
+        hr_2 = np.zeros((self._num_modes, self._num_modes))
+
+        for i in range(self._num_modes):
+            hr_1[i, i] = 2 * h_1[i, i] + h_2[i, i, i, i]
+
+            for j in range(self._num_modes):
+                if i != j:
+                    hr_1[i, j] = h_2[i, i, j, j]
+                    hr_2[i, j] = 2 * h_2[i, j, j, i] - h_2[i, j, i, j]
+
+        return hr_1, hr_2
+
+
+def second_q_ops_electronic_energy(self: ElectronicEnergy) -> ListOrDictType[VibrationalOp]:
+    """Creates the second quantization operators in the restricted formalism
+    arxiv: 2002.00035, Equation (2)
+    https://arxiv.org/pdf/2002.00035.pdf
+    """
+
+    hr_1 = self.get_electronic_integral(ElectronicBasis.MO, 1).get_matrix()
+    hr_2 = self.get_electronic_integral(ElectronicBasis.MO, 2).get_matrix()
+
+    num_modes = hr_1.shape[0]
+
+    label_m = lambda x: "I" * x + "-" + "I" * (num_modes - x - 1)
+    label_p = lambda x: "I" * x + "+" + "I" * (num_modes - x - 1)
+
+    # Create pair creation and annihilation operators for the spatial orbitals
+    b = [VibrationalOp(label_m(i), num_modes=num_modes, num_modals=1) for i in range(num_modes)]
+    b_dag = [VibrationalOp(label_p(i), num_modes=num_modes, num_modals=1) for i in range(num_modes)]
```

Comment:
You should be able to use sparse labels here:
```suggestion
    # Create pair creation and annihilation operators for the spatial orbitals
    b = [VibrationalOp(f"-_{i}", num_modes=num_modes, num_modals=1) for i in range(num_modes)]
    b_dag = [VibrationalOp(f"+_{i}", num_modes=num_modes, num_modals=1) for i in range(num_modes)]
```

Reply:
I have not used sparse labels for the following reasons:
* For the `VibrationalOp` the sparse label has the notation `-_{mode_index}*{modal_index}` and I think it is not really nice to explicitly state the modal index when we do not use the modal functionality of the `VibrationalOp`.
* Using labels like `+_0*0` triggers the warning `Number of raising and lowering operators do not agree for mode 0 in label +_0*0` as the `VibrationalOp` would like to conserve the number of excitations in each mode. The warning is not triggered when I specify the label explicitly.

I will add a comment in the code to clarify why I did not use sparse notation.